lib/src/_shared/KeyboardDateInput.tsx

```diff
@@ -91,6 +93,8 @@ export const KeyboardDateInput: React.FC<DateInputProps> = ({
 
   const adornmentPosition = InputAdornmentProps?.position || 'end';
   const inputProps = {
+    ref: containerRef,
+    inputRef: forwardedRef,
```

Comment:
Something is off with the forwardRef logic, the standard we went for is to apply it on the root element, here, it seems to be forwarded on an arbitrary deeper element: the text box. Is it correct?

Reply:
This sounds like a good reason to start using https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/test-utils/describeConformance.js in the tests, it will increase the consistency